Lib/tkinter/filedialog.py

```diff
@@ -385,9 +385,27 @@ def askopenfilename(**options):
 
 
 def asksaveasfilename(**options):
-    "Ask for a filename to save as"
+    """Ask for a filename to save as"""
+
+    # If no default extension is provided, set it to the first filetype
+    if "defaultextension" not in options:
+        filetypes = options.get("filetypes")
+        if filetypes:
+            first_ext = filetypes[0][1]  # Get "*.txt" from ("Text files", "*.txt")
+            if first_ext.startswith("*"):
+                options["defaultextension"] = first_ext[1:]  # ".txt"
```

Comment:
This will work incorrectly in cases like "*-*.txt" or "*.a[0-9]". It is easier to just specify an unambiguous `defaultextension` argument in your code.

This change also removes possibility to enter extensionless name, which may be desirable.